libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/android/AndroidGradleWrapper.groovy

```diff
@@ -132,8 +132,13 @@ class AndroidGradleWrapper {
               && variantData.scope.getRenderscriptCompileTask() != null) {
         result.add(variantData.scope.getRenderscriptSourceOutputDir());
       }
+    }
 
-      return result
+    def getExtraSourcesMethod = variantData.getMetaClass().getMetaMethod("getExtraGeneratedSourceFolders")
+    if (getExtraSourcesMethod.returnType.metaClass == List.metaClass) {
+      result.addAll(variantData.getExtraGeneratedSourceFolders())
```

Comment:
Returning null on something that returns a collection. Those developers should have their programming license revoked.

Good catch, will update tomorrow.